### DISCUSSION TITLE
mock Jigsaw in an external package for feature testing

### DISCUSSION THREAD

**Discussion by medilies** [tighten/jigsaw#663] [Q&A]
My package [jigsaw-localization](https://github.com/elaborate-code/jigsaw-localization) offers a façade that handles loading **localization data** into the `config` on the `beforeBuild` event.

I need to get a Jigsaw instance, set the config on it, then use my helpers and assert some output.

I don't know how to hook to Jigsaw's lifecycle and use its container.

**Comment by bakerkretzmar** [comment]
That's a really cool package!

Great question—my suggestion would be to create snapshot tests and just run Jigsaw on them for real and make sure the output is what you expect. I think you could probably set them up similarly to Jigsaw's own snapshot tests: https://github.com/tighten/jigsaw/blob/main/tests/SnapshotsTest.php. Since `tightenco/jigsaw` is already installed in your package, you can run its build command anywhere you want.

There might be a way to 'mock' Jigsaw and hook into the container directly but I think snapshot tests are going to be a lot simpler and easier to set up, and also provide a better end-to-end guarantee that everything works the way you want in a real site.

Let me know how it goes and if you have other questions!

**Reply by medilies** [reply]
Thank you for the great suggestions. I make sure to try the snapshot approach as soon as possible!
